### PR TITLE
chore: setup v5 release channel

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

Setup Gatsby v5 alpha channel. It's similar to our v4 release. We merge everything in master behind `_CFLAGS_.MAJOR`, this allows us to make breaking changes when publishing v5 but keeping v4 channel working.

We should use `_CFLAGS_.MAJOR` for all things that's javascript based. For other related files we can use `git patches` that auto apply in order.

### CODE REVIEW

**Comment by LekoArts** [comment]
This here also needs an update, right?

https://github.com/gatsbyjs/gatsby/blob/7fe8e511b0dcd1707eefca2c63b0f27174d7252c/packages/babel-preset-gatsby-package/lib/index.js#L14

**Comment by LekoArts** [comment]
@TylerBarnes What about https://github.com/gatsbyjs/gatsby/blob/7fe8e511b0dcd1707eefca2c63b0f27174d7252c/packages/gatsby-source-wordpress/src/utils/gatsby-version.ts, can this be changed to just behave normal now?

**Comment by wardpeet** [comment]
The above we want to keep as is; it's the default that we run with.

File: patches/v5/0-gatsby-peerdeps.patch [hunk 1]
```diff
@@ -1,1169 +1,1107 @@
...
    "main": "index.js",
    "peerDependencies": {
      "cxs": ">=5.0.0",
--    "gatsby": "^3.0.0-next.0",
-+    "gatsby": "^4.0.0-alpha-9689ff",
-     "react": "^16.9.0 || ^17.0.0",
-     "react-dom": "^16.9.0 || ^17.0.0"
+-    "gatsby": "^4.0.0-next",
++    "gatsby": "^5.0.0-alpha-v5",
+     "react": "^16.9.0 || ^17.0.0 || ^18.0.0",
```
Comment (LekoArts) [inline patches/v5/0-gatsby-peerdeps.patch:76]: Shouldn't we only do React 18?

**Review by LekoArts — APPROVED** [review]

**Pull request merged**